### DISCUSSION TITLE
Profiles valid configuration files paths

### DISCUSSION THREAD

**Discussion by kattouf** [jdx/mise#2168] [Q&A]
Hello! I tried to use the `Profiles` feature based on [doc](https://mise.jdx.dev/profiles.html) and didn't understand which env-dependent files I could use:

It is clearly stated at the beginning of the document:
> To enable, set MISE_ENV to an environment like development or production. mise will then look for a .mise.{MISE_ENV}.toml file in the current directory.

So for now I think I can only use one type of paths: `.mise.{MISE_ENV}.toml`

But below we can see the priority list, which contains many more paths depending on MISE_ENV:
> The priority of these files goes in this order (bottom overrides top):
    ...
    - .config/mise/config.{MISE_ENV}.toml
    - mise/config.{MISE_ENV}.toml
    - mise.{MISE_ENV}.toml
    - .mise/config.{MISE_ENV}.toml
    - .mise.{MISE_ENV}.toml
    - .config/mise/config.{MISE_ENV}.local.toml
    - mise/config.{MISE_ENV}.local.toml
    - .mise/config.{MISE_ENV}.local.toml
    - .mise.{MISE_ENV}.local.toml

And when I tried to use `.mise/config.{MISE_ENV}.toml` or `mise/config.{MISE_ENV}.toml` it didn't have any effect:
(this is my project preferred structure)

I created a small example to reproduce what I want, where:
CI Environment: should use homebrew python 3.12
Dev Environment: should install python 3.10 
p.s. Python versions different to test the `MISE_ENV` call using `python3 --version`

#### Example with `mise/config.{MISE_ENV}.toml` - doesn't work

<img width="554" alt="image" src="https://github.com/jdx/mise/assets/7829589/024c1d47-0db4-400e-bdcb-f7c51ec9b344">

As you can see, the Python version is the same and equal to the Python homebrew version.

#### Example with `.mise.{MISE_ENV}.toml` - work

<img width="610" alt="image" src="https://github.com/jdx/mise/assets/7829589/a0b81437-057f-4fb6-982b-60db612cf102">

Here dev python is 3.10 and ci is 3.12, just like we wanted

#### So my question is: where is the problem here?
1. I have misunderstood something and then I would like you to point out to me what I am doing wrong.
2. The document contains misinformation about available paths depending on the environment.
3. The document is correct, but there is a bug or no implementation at all.

**P.S. I'm glad the configuration from root works, but I'd just like to use other methods if it's legal.**

**Comment by jdx** [comment]
I'm a bit busy at the moment but it might be worth glancing at the code for this and seeing if I missed something obvious: https://github.com/jdx/mise/blob/d7f64e23f8d95a10fbd0eaccdd9075841c1936c6/src/config/mod.rs#L460-L500

**Comment by kattouf** [comment]
I've analyzed all configuration sequence declarations and seem to have found inconsistencies that lead to unexpected behavior:

I took 3 sources:
1. [Configuration doc](https://mise.jdx.dev/configuration.html)
2. [Profiles doc](https://mise.jdx.dev/profiles.html)
3. [Source code](https://github.com/jdx/mise/blob/d7f64e23f8d95a10fbd0eaccdd9075841c1936c6/src/config/mod.rs#L460-L500)

Then have brought them to a single format and sorting order for ease of comparison, after which, divided them into logical groups:
1. Original configs
2. Local configs
3. Profiles configs
4. Local + Profiles configs

Then take configuration order from [Configuration doc](https://mise.jdx.dev/configuration.html) as order reference and began to compare.
If I took the reference correctly and understood the original author’s wishes, then as a result we have the following inconsistencies between the documents and the implementation:

<img width="1452" alt="mise_configs_invesigation" src="https://github.com/jdx/mise/assets/7829589/b3c5aec1-de7b-4f4a-a726-6520be583141">

**Comment by kattouf** [comment]
Partially fixed [here](https://github.com/jdx/mise/releases/tag/v2024.5.22)
